src/main/java/com/exasol/dbbuilder/dialects/Table.java

```diff
@@ -84,8 +73,20 @@ public Table insert(final Object... values) {
                     "Column count mismatch. Tried to insert row with " + values.length + " values into table \""
                             + this.getFullyQualifiedName() + "\" which has " + this.columns.size() + " columns");
         }
-        this.rows.add(Arrays.asList(values));
-        this.writer.write(this, values);
+        final List<Object> row = Arrays.asList(values);
+        this.writer.write(this, Stream.of(row));
+        return this;
+    }
+
+    /**
+     * Insert multiple rows at once. Compared to inserting each row using {@link #insert(Object...)} this is a lot
+     * faster.
+     * 
+     * @param rows stream of rows to insert
+     * @return {@code this} for fluent programming
+     */
+    public Table bulkInsert(final Stream<List<Object>> rows) {
```

Comment:
It does not have early column size check as the other one.

Reply:
I can't validate here all rows before, since for that I would have to store all rows in memory at once. Do you think I should validate it anyway? It would give a better error message. But when the exception is thrown some rows would already be written.

Reply:
Ahh, I see, this would require to consume the stream. 

It would already throw if they columns do not match the prepared statement placeholders (lines 93-95 in immediate object writer). But, you would have to rollback to remove the partial insert. I think you could do this when catching exception.

Could you please test it?

Reply:
As discussed, throw an exception and indicate there may be partial writes in message.